ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/infraestructure/antlr/visitors/method/JavaMethodVisitor.java

```diff
@@ -0,0 +1,131 @@
+import edu.usb.argos.ASTProcessor.visitor.core.entities.method.MethodInformation;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.method.ParameterInformation;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.IExpressionCollector;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.IModifierCollector;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.IStatementCollector;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.nodes.Expression;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.nodes.Statement;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.nodes.Token;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.visitor.IMethodAnalyzerVisitor;
+import edu.usb.argos.ASTProcessor.visitor.infraestructure.antlr.adapters.AntlrExpressionAdapter;
+import edu.usb.argos.ASTProcessor.visitor.infraestructure.antlr.adapters.AntlrStatementAdapter;
+import edu.usb.argos.ASTProcessor.visitor.infraestructure.antlr.adapters.AntlrTokenAdapter;
+import edu.usb.argos.ASTProcessor.visitor.shared.validation.ContextValidator;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.ParserRuleContext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@EqualsAndHashCode(callSuper = true)
+@Value
+public class JavaMethodVisitor extends
+        JavaParserBaseVisitor<MethodInformation<JavaParser.StatementContext, JavaParser.ExpressionContext, CommonTokenStream>>
+        implements IMethodAnalyzerVisitor<ParserRuleContext,
+        MethodInformation<JavaParser.StatementContext, JavaParser.ExpressionContext, CommonTokenStream>> {
+
+    CommonTokenStream tokenStream;
+    IStatementCollector<JavaParser.StatementContext, JavaParser.MethodDeclarationContext> statementCollector;
+    IExpressionCollector<JavaParser.ExpressionContext, JavaParser.MethodDeclarationContext> expressionCollector;
+    IModifierCollector<ParserRuleContext> modifierCollector;
+    private static final String EMPTY_RETURN_TYPE = "";
+
+    @Override
+    public MethodInformation<JavaParser.StatementContext, JavaParser.ExpressionContext, CommonTokenStream> visitMethod(ParserRuleContext ctx) {
+        return ContextValidator.validateAndExecute(
+                ctx,
+                JavaParser.MethodDeclarationContext.class,
+                this::buildMethodInfo,
+                null
+        );
+    }
+
+    @Override
+    public List<String> getMethodModifiers(ParserRuleContext ctx) {
+        return modifierCollector.collectModifiers(ctx);
+    }
+
+    @Override
+    public String getReturnType(ParserRuleContext ctx) {
+        return ContextValidator.validateAndExecute(
+                ctx,
+                JavaParser.MethodDeclarationContext.class,
+                this::extractReturnType,
+                EMPTY_RETURN_TYPE
```

Comment:
I think it would be good if instead of putting an empty string you could do it with the Optional object which would be much better, but if you think it is not good that way could you tell me?

Reply:
While using Optional is a good practice for null handling, changing the interface to return Optional<String> would add unnecessary complexity and force all consumers to handle Optional. So I am now making a balance between handling nulls using Optional and keeping the interface simple and straightforward without affecting consumers.